src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -686,6 +760,67 @@ def cmac_algorithm_supported(self, algorithm):
     def create_cmac_ctx(self, algorithm):
         return _CMACContext(self, algorithm)
 
+    def sign_x509_request(self, builder, private_key, algorithm):
+        # TODO: check type of private key parameter.
```

Comment:
We decided to keep the general mechanism but this name still doesn't seem appropriate. What you do is create a CSR, not just sign it, so I'd call it create_x509_request().